hack/install-go.sh

```diff
@@ -2,7 +2,8 @@
 
 destination=$1
 version=$(curl -s https://go.dev/dl/?mode=json | jq -r ".[0].version")
-tarball=$version.linux-amd64.tar.gz
+arch=$(uname -m | sed 's/x86_64/amd64/;s/aarch64/arm64/')
```

Comment:
Is that required? IIUC the binaries for all platforms are cross-compiled on amd64?

Reply:
Yes.that's correct.

Reply:
If you're asking why we use the `sed` command, here's the explanation:
The terms "x86_64" and "amd64" refer to the same architecture. However, the `uname -m` command returns "x86_64" to describe this architecture. On the other hand, the Go programming language uses the term "amd64" for this architecture. Therefore, to ensure consistency with Go and cross-compilation setups, we use the `sed` command to rename "x86_64" to "amd64".

Reply:
Yes, I got that. Just asking if we need this change or if we can continue to cross compile on amd64.